Source/WebCore/style/StyleBuilderCustom.h

```diff
@@ -2040,6 +2044,30 @@ inline void BuilderCustom::applyValueStrokeColor(BuilderState& builderState, CSS
     builderState.style().setHasExplicitlySetStrokeColor(true);
 }
 
+inline void BuilderCustom::applyValueColor(BuilderState& builderState, CSSValue& value)
+{
+    auto& primitiveValue = downcast<CSSPrimitiveValue>(value);
+
+    // For the color property, current color is actually the inherited computed color.
+    auto absoluteColorOrInheritColor = [&](const StyleColor& color) {
+        if (color.isCurrentColor()) {
+            auto& parentStyle = builderState.parentStyle();
+            return parentStyle.color();
+        }
+        return color;
+    };
```

Comment:
Here we resolve currentcolor for color property, making the other bits in the patch where it is resolved again unnecessary.

This sort of point out that the color and visitedLinkColor should remain as Color instead of StyleColor. That would clean up other parts of the patch.